src/omnijson/OmniJson/OmniJsonObject.cs

```diff
@@ -0,0 +1,118 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace JsonLD.OmniJson
+{
+    public class OmniJsonObject : OmniJsonToken, IDictionary<string, OmniJsonToken>
+    {
+        IDictionary<string, object> _wrapped;
+
+        public override object Unwrap() => _wrapped;
+
+        public OmniJsonObject(IDictionary<string,object> dict)
+        {
+            _wrapped = dict;
+        }
+
+        public OmniJsonObject()
+        {
+            _wrapped = new Dictionary<string,object>();
+        }
+
+        public bool IsEmpty() => _wrapped.Count == 0;
+
+        public override OmniJsonToken this[object index] { get => this[(string)index]; set => this[(string)index] = value; }
+
+        public virtual OmniJsonToken this[string key] { get => _wrapped.ContainsKey(key) ? Wrap(_wrapped[key]) : null; set => _wrapped[key] = value?.Unwrap(); }
+
+        public virtual ICollection<string> Keys => throw new NotImplementedException();
```

Comment:
Better to not implement `IDictionary` than to throw `NotImplementedException`, no?